main

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env ruby
+require 'date'
+
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+# rubocop: disable Metrics/ClassLength
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/AbcSize(RuboCop)
+
+# Public: App class used to run main logic.
+#
+# All methods are class methods and should be called on the App class.
+#
+# Examples
+#
+# app = App.new
+# app.run
+class App
+  # Puclic: Initialize  App class
+  #
+  # Returns: App Object
+  def initialize
+    @book = []
+    @person = []
+    @rental = []
+  end
+
+  # Public: Main method used to run the app.
+  def run
+    puts 'Loading App...'
+    sleep 0.8
+    menu
+  end
+
+  def menu
+    puts `clear`
+    puts "Welcome to Quest Student Library!\n\n"
+    puts 'What would you like to do?'
+    puts 'Please choose an option by entering a number:'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a book'
+    puts '5. Create a rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Exit'
+
+    input = gets.chomp.to_i
+
+    validate_input input
+  end
+
+  def validate_input(input)
+    case input
+    when 1
+      list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_a_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      list_rentals_by_person_id
+    when 7
+      puts `clear`
+      puts 'Thank you for using Quest Student Library. See you next time!'
+      sleep 1.5
+      exit
+    else
+      puts `clear`
+      puts 'Invalid input! Please enter a number between 1 and 7.'
+      sleep 1
+      menu
+    end
+  end
+
+  def create_a_person
+    print 'Do you want to create a student (1) or teacher (2)? [Input the number]: '
+    input = gets.chomp.to_i
+
+    case input
+    when 1
+      create_a_student
+    when 2
+      create_a_teacher
+    else
+      puts 'Invalid input! Kindly enter a number between 1 and 2.'
+    end
+  end
+
+  def list_all_books
+    puts `clear`
+    puts "\n\n\n\t\t     ALL AVAILABLE BOOKS \n\n\n"
+    puts 'There are no books in the library. Please add some books.' if @book.empty?
+    @book.each do |book|
+      puts "#{book.title} by #{book.author}"
+    end
+    puts 'Press any key to go back to the main menu....'
+    gets.chomp
+    menu
+  end
```

Comment:
- [ ] It would be better if we move the functionalities into another file, for example; app.rb. These methods should hold all functionalities, could be initialize in the main or entry point and each methods used respectively. This [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point) can helps us out
```
**app.rb
class App
.....
end
**main.rb
def main
app = App.new()
end
```